src/node_worker.cc

```diff
@@ -388,9 +407,12 @@ Worker::~Worker() {
   Mutex::ScopedLock lock(mutex_);
   JoinThread();
 
-  CHECK(stopped_);
+  CHECK(thread_stopper_ == nullptr || thread_stopper_->IsStopped());
   CHECK(thread_joined_);
 
+  delete thread_stopper_;
+  delete on_thread_finished_;
```

Comment:
If we free these from the constructor anyway, maybe we could make them `std::unique_ptr`s so that that happens automatically?

Reply:
done, converted into unique ptrs, and constructed using `reset` primitive.